queueConversationControl/queueConversationControl.js

```diff
@@ -0,0 +1,187 @@
+            query.find('td.ewt').text(queueData.estimatedWaitTimeForNewConversations);
+            query.find('td.serviceLevel').text(queueData.serviceLevelPercentageToday);
+            query.find('td.offered').text(queueData.offeredConversationsToday);
+            query.find('td.answered').text(queueData.answeredConversationsToday);
+        });
+    }
+}
+
+function populateQueueConversationStats(conversationData) {
+    if (conversationData === undefined || conversationData === null) {
+        console.error('populateQueueConversationStats -', conversationData);
+        return;
+    }
+    var matchingConversationRows = $(`[id='${conversationData.conversationId}']`);
+    if (!matchingConversationRows) {
+        console.warn('No rows found for conversation with id %s', conversationData.conversationId);
+    } else {
+        matchingConversationRows.each(function (i, conversationRow) {
+            var query = $(conversationRow);
+            query.attr('id', conversationData.id);
+            query.find('td.conversationId').text(conversationData.conversationId);
+            query.find('td.mediaType').text(conversationData.mediaType);
+            query.find('td.caseId').text(conversationData.caseId);
+            query.find('td.fromName').text(conversationData.fromName);
+            query.find('td.fromAddress').text(conversationData.fromAddress);
+            query.find('td.toName').text(conversationData.toName);
+            query.find('td.toAddress').text(conversationData.toAddress);
+            query.find('td.subject').text(conversationData.subject);
+            query.find('td.timeOfferedToSystem').text(conversationData.timeOfferedToSystem);
+        });
+    }
+}
+
+function addQueueRow(searchTerm) {
+    $('#queueId').val('');
+    micc.getQueueStats(searchTerm, function (queueData) {
+        var queueRow = $(`<tr id="${queueData.id}" class="queueRow">
+            <td><button class="w3-btn w3-round-xxlarge w3-tiny w3-blue" title="Remove"
+                onclick="deleteQueue(this, '${queueData.id}');">X</button></td>
+            <td class="name">?</td>
+            <td class="reporting">?</td>
+            <td class="mediaType">?</td>
+            <td class="waitingConversations">?</td>
+            <td class="longestWaiting">?</td>
+            <td class="ewt">?</td>
+            <td class="serviceLevel">?</td>
+            <td class="offered">?</td>
+            <td class="answered">?</td>cls
+        </tr>`)
+            .appendTo('#queueStatsTable > tbody:last-child');
+        miccSignalr.addQueueStateMonitor([queueData.id]);
+        miccSignalr.addQueueConversationsMonitor([queueData.id]);
+        populateQueueStats(queueData);
+        micc.getQueueConversations(queueData.id, function (queueConversations) {
+            addQueueConversationRows(queueConversations._embedded.items);
+        });
+    });
+}
+
+function addQueueConversationRows(conversations) {
+    for (var i = 0; i < conversations.length; i++) {
```

Comment:
Use the .each function instead. If you don't care about the index, it's more appropriate. Apply to following functions.